app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,35 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
+    @posts = @user.recent_posts
   end
 
   def show
     @user = User.find(params[:user_id])
     @post = @user.posts.find(params[:id])
-    @comments = @post.comments.all
+    @comments = @post.comments.all.order(created_at: :desc)
+  end
+
+  def new
```

Comment:
There should be a pair of routes in the `config/routes.rb` file for creating posts. The user is unable to create new posts as there is no route to the controller.